pkg/tests/testsupport/metric_loader.go

```diff
@@ -0,0 +1,244 @@
+package testsupport
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/prompb"
+	"golang.org/x/time/rate"
+)
+
+// PromLoader is responsible for loading Prometheus data files and providing iterator over Prometheus data
+type PromLoader interface {
+	Iterator() PromIterator
+	Close() error
+}
+
+type promLoader struct {
+	db     *tsdb.DBReadOnly
+	blocks []tsdb.BlockReader
+}
+
+// PromIterator allows us to iterate over Prometheus data
+type PromIterator interface {
+	Next() bool
+	Get() timeSeries
+}
+
+type timeSeries struct {
+	seriesHash uint64
+	Val        prompb.TimeSeries
+}
+
+// Sample loaded from Prometheus block
+type BlockSample struct {
+	ts       int64
+	val      float64
+	lblsHash uint64
+}
+
+type promIterator struct {
+	blocks       []tsdb.BlockReader
+	curBlockIdx  int
+	labelsCache  map[uint64]labels.Labels
+	blockSamples []*BlockSample
+	curSampleIdx int
+}
+
+var allMatcher = labels.MustNewMatcher(labels.MatchRegexp, "", ".*")
+
+// Next moves to next sample
+func (i *promIterator) Next() bool {
+	if i.curSampleIdx == -1 && i.curBlockIdx == -1 {
+		i.curBlockIdx++
+		if err := i.loadBlockSamples(); err != nil {
+			panic(err)
+		}
+		i.curSampleIdx = 0
+		return true
+	}
+	if i.curSampleIdx < len(i.blockSamples)-1 {
+		i.curSampleIdx++
+		return true
+	}
+	if i.curBlockIdx == len(i.blocks)-1 {
+		return false
+	}
+	i.curBlockIdx++
+	if err := i.loadBlockSamples(); err != nil {
+		panic(err)
+	}
+	i.curSampleIdx = 0
+	return true
+}
+
+func (it *promIterator) loadBlockSamples() error {
+	log.Info("loading blocks, total samples", it.blocks[it.curBlockIdx].Meta().Stats.NumSamples, "series", it.blocks[it.curBlockIdx].Meta().Stats.NumSeries)
```

Comment:
```suggestion
	log.Infof("loading blocks, total samples: %v, series: %v",
			it.blocks[it.curBlockIdx].Meta().Stats.NumSamples,
			it.blocks[it.curBlockIdx].Meta().Stats.NumSeries)
```
nit: a bit easier to read in code review view.